functions/src/routes/express_portal.ts

```diff
@@ -34,5 +35,38 @@ app_portal.get("/auth0/applications", get_active_applications);
 app_portal.get("/auth0/discord", get_user_metadata);
 app_portal.post("/auth0/verify-discord", verify_in_acm_server);
 
+// Create Vanity Link
+app_portal.post(
+  "/auth0/vanity/create",
+  [
+    requireField({
+      fieldName: "first_name",
```

Comment:
Just curious: is there any way to make this more type-safe? It appears that modifying any of these `requireField` calls may break functionality within the function itself.

Reply:
I am not sure what you mean here because I believe that I specified the required parameters for `requireField` here: https://github.com/acmutd/portal-backend/blob/vanity-link/functions/src/services/validate-request.ts#L13

If my interpretation is correct, I only need to do this: https://github.com/acmutd/portal-backend/pull/110#discussion_r721399410